### ISSUE TITLE
Debian package cannot be purged if elasticsearch user does not exist in local database

### ISSUE THREAD

**Issue by Nihlus** [elastic/elasticsearch#94795]
### Elasticsearch Version

8.6.2

### Java Version

_bundled_

### OS Version

Linux 5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux

### Problem Description

When attempting to purge the elasticsearch debian package on a system that does not have an elasticsearch user in the local user database, the purge fails. 

In my case, I manage the elasticsearch user in an LDAP database (for SAN filesystem permissions), and as such it is not present in the local user database of the machine. Typically, packages would not attempt to remove a user that does not already exist, treating it as a success if the user has already been removed.

### Steps to Reproduce

1. Install elasticsearch (apt install elasticsearch)
2. Uninstall elasticsearch (apt remove elasticsearch)
3. Delete the elasticsearch user (userdel elasticsearch)
4. Purge elasticsearch (apt purge elasticsearch)
5. Observe failure

**Comment by mark-vieira** [comment]
We [currently check](https://github.com/elastic/elasticsearch/blob/d2dbef506396a6eeae09f9e028de8c95d5b2bc6a/distribution/packages/src/common/scripts/postrm#L130) via `id` whether the user exists. It seems like this is going to return a positive result even for users that don't exist in the local passwd file. I suspect we should continue to use `id` to determine whether to _create_ the user but perhaps move to use `getent passwd` to determine whether we should remove the user. @rjernst do you have any objection to making that change?

**Comment by rjernst** [comment]
Using something like `getent passwd` to check if the user exists locally sounds like a good approach.

**Comment by Nihlus** [comment]
I've been using that as a local workaround; however, it's important to use the `-s` flag and explicitly limit the check to the local files (otherwise, it'll still report back with the LDAP user). This goes for the group as well.

**Comment by mark-vieira** [comment]
> I've been using that as a local workaround; however, it's important to use the `-s` flag and explicitly limit the check to the local files (otherwise, it'll still report back with the LDAP user). This goes for the group as well.

@Nihlus do you similarly use LDAP for managing the `elasticsearch` group as well? From what I can tell we already use `getent group` there. Would you still run into the same issue as via `id` though, that by default it'll lookup database entries via whatever services are configures in `/etc/nsswitch.conf` unless we explicitly something like `-s files`?

**Comment by Nihlus** [comment]
Yep, that's correct. Both the user and group for `elasticsearch` (and `kibana`, as it happens) are managed via LDAP. `getent group` follows nsswitch.conf by default, which means that it gets its data from the LDAP server and is satisfied with that. The `-s files:group` flag forces it to limit its search to the local files which we can control.

With that said, packages generally don't try to delete created users and groups on removal. The Debian policy hasn't put its foot down on whether to do it or not, but Fedora explicitly recommends against it. There's some good reading on the topic [here](https://wiki.debian.org/AccountHandlingInMaintainerScripts), [here](https://fedoraproject.org/wiki/Packaging:UsersAndGroups), and [here](https://unix.stackexchange.com/questions/47880/how-debian-package-should-create-user-accounts).

The suggestion in the first link (making account/group removal a non-failing operation even if the account couldn't be removed or wasn't found) is also a workable option if it remains desirable to remove the account/group on package purge.
